src/robot/reporting/xunitwriter.py

```diff
@@ -90,3 +91,9 @@ def visit_errors(self, errors):
 
     def end_result(self, result):
         self._writer.close()
+
+    def _starttime_to_isoformat(self, stime):
+        isoformat = None
+        if stime:
+            isoformat = f'{stime[:4]}-{stime[4:6]}-{stime[6:8]}T{stime[9:22]}000'
+        return isoformat
```

Comment:
What happens if `None` is returned here? Will we get `timestamp='None'`, not `timestamp` at all, or a crash? I hope it's the middle one but test should be added for it.

I'd also write this code like below to simplify it a bit:
```
if not stime:
    return None
return f'...'
```

Reply:
```return None``` yields ```timestamp=""``` currently. I made that part unit testing by hand, but will write the test as well. Also, I'll modify the routine more simply as proposed.